v22.1/security-reference/scram-authentication.md

```diff
@@ -0,0 +1,326 @@
+](#enabling-scram-sha-256-authentication-for-new-users-roles).
+
+Required cluster settings:
+
+
+#### `server.user_login.password_encryption`
+
+Enable SCRAM for new passwords.
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+SET CLUSTER SETTING server.user_login.password_encryption = 'scram-sha-256';
+~~~
+
+#### `server.user_login.store_client_pre_hashed_passwords.enabled`
+
+Enable the CockroachDB cluster to accept pre-computed SCRAM-SCHA-256 salted password hashes, rather than accepting a plaintext password and computing the hash itself.
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+SET CLUSTER SETTING setting server.user_login.store_client_pre_hashed_passwords.enabled = true;
+~~~
+
+#### `server.host_based_authentication.configuration`
+
+Allow *only* `scram-sha-256` (not the weaker `password`) in your [Authentication Configuration (HBA)](authentication.html#authentication-configuration).
+
+{{site.data.alerts.callout_danger}}
+After this change, users will not be able to sign in with plaintext passwords. All critical SQL users must be migrated to SCRAM-SHA-256 authentication before making this change:
+1. All client drivers must be updated to a version that supports SCRAM handshakes.
+2. All stored credentials for client apps must be updated to use SCRAM encoded passwords. This procedure is detailed in [Migrating existing users/roles to SCRAM-SHA-256 authentication](#migrating-existing-users-roles-to-scram-sha-256-authentication)
+{{site.data.alerts.end}}
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+SET CLUSTER SETTING server.host_based_authentication.configuration TO '
+# TYPE    DATABASE  USER          ADDRESS     METHOD
+# any other higher-priority rules ...
+  host    all       human         all         scram-sha-256
+  host    all       app_user      all         cert
+  host    all       cyborg        all         cert-scram-sha-256
+# any other lower-priority rules ...
+~~~
+
+### Handling plaintext credentials securely
+
+Using SCRAM authentication, a CockroachdDB cluster need not store or handle the plaintext passwords used by clients to authenticate. However, to achieve full separation of concerns, the plaintext credentials must be isolated so that they are exposed as little as possible in transit, storage, and during computation.
+
+Generally, this can be best achieved by handling plaintext credentials only in a dedicated credential management environment, such as a secure compute instance or secure physical workstation.
+
+That environment must be provisioned with:
+
+- A tool for [creating PostgreSQL/CockroachDB-formatted SCRAM-SHA-256 hashes](#managing-users-with-pre-hashed-passwords), such as the example script provided in the [appendix](#appendix-python-scram-hashing-script).
+- A CRDB client with ['USER CREATE/ALTER privileges'](../create-user.html#create-a-user-that-can-create-other-users-and-manage-authentication-methods-for-the-new-users)
```

Comment:
This second bullet doesn't have a period at the end, while the other bullets do have a period.